src/main/java/org/jabref/gui/entryeditor/citationrelationtab/RelatedEntriesCache.java

```diff
@@ -0,0 +1,25 @@
+package org.jabref.gui.entryeditor.citationrelationtab;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.identifier.DOI;
+
+public class RelatedEntriesCache {
+    private final Map<String, List<BibEntry>> RELATED_ENTRIES_MAP = new HashMap<>();
+
+    public List<BibEntry> lookupRelatedEntries(BibEntry entry) {
+        return RELATED_ENTRIES_MAP.getOrDefault(entry.getDOI().map(DOI::getDOI).orElse(""), Collections.emptyList());
+    }
+
+    public void cacheOrMerge(BibEntry entry, List<BibEntry> relatedEntries) {
+        entry.getDOI().ifPresent(doi -> RELATED_ENTRIES_MAP.putIfAbsent(doi.getDOI(), relatedEntries));
```

Comment:
The signature name is a bit misleading? -- `upateIfNotExists` is more fitting? -- Or do you really want to merge? Then, the could should look differently.

Reading `List<BibEntry> references = fetcher.searchCiting(entry); cache.cacheOrMergeReferences(entry, references);`, I would assume that you intend a merge - or even a replacement (which is the opposite of the current code)

Reply:
Hmmm, I don't remember what I thought back then. Anyway, merging doesn't make sense here since we want the most up-to-date version of references/citations so merging cached and newly fetched references doesn't help. `cacheOrReplace` would be more appropriate in this case.